lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -302,7 +302,7 @@ public function getAST()
      */
     public function match($token)
     {
-        $lookaheadType = $this->lexer->lookahead['type'];
+        $lookaheadType = $this->lexer->lookahead['type'] ?? null;
```

Comment:
So lookahead is not an array in your case?

Reply:
Yep, `$this->lexer->lookahead` [is null](https://github.com/doctrine/lexer/blob/1.2.0/lib/Doctrine/Common/Lexer/AbstractLexer.php#L59) when executing the CLI command I posted in the intro. Is this a bug in itself?

Reply:
Not according to the typehint, it is `array|null` - so this change is right.

Reply:
Yes, but it could have been nullable only under certain conditions, that could be assumed would never be met at that point. Anyway if this is the expected behaviour, then yes the PR just fixes the notice on PHP 7.4.